File: Classes/ViewControllerBase.m
//
//  Copyright © 2018 SpotX, Inc. All rights reserved.
//

#import <SpotX/SpotX.h>
#import "ViewControllerBase.h"

@interface ViewControllerBase ()

@end

@implementation ViewControllerBase

- (void)viewDidLoad {
    [super viewDidLoad];
  
  // update version label
  _versionLabel.text = [NSString stringWithFormat:@"VERSION %@", [SpotX version]];
  
  // create "done" button on keyboard
  _keyboardDoneButtonView = [[UIToolbar alloc] init];
  [_keyboardDoneButtonView sizeToFit];
  UIBarButtonItem *doneButton = [[UIBarButtonItem alloc] initWithTitle:@"Done " style:UIBarButtonItemStylePlain target:self action:@selector(doneClicked:)];
  UIBarButtonItem *fakeButton = [[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemFlexibleSpace target:nil action:nil];
  [_keyboardDoneButtonView setItems:[NSArray arrayWithObjects:fakeButton, doneButton, nil]];
}

- (void)doneClicked:(id)sender {
  [self dismissKeyboard];
}

- (void)backgroundTap:(UITapGestureRecognizer *)sender {
  [self dismissKeyboard];
}

- (void)dismissKeyboard {
}

-(void)showMessage:(NSString *)message {
  UIAlertController * alert = [UIAlertController alertControllerWithTitle:nil message:message preferredStyle:UIAlertControllerStyleAlert];
  
  UIAlertAction* defaultAction = [UIAlertAction actionWithTitle:@"OK" style:UIAlertActionStyleDefault handler: ^(UIAlertAction * action) {}];
  [alert addAction:defaultAction];
  [self presentViewController:alert animated:YES completion:nil];
}

@end

File: Classes/ViewControllerBase.swift
//
//  Copyright © 2018 SpotX. All rights reserved.
//

import UIKit
import SpotX

class ViewControllerBase: UIViewController {
  
  @IBOutlet weak var _versionLabel: UILabel?
  
  var keyboardDoneButtonView: UIToolbar!
  
  override func viewDidLoad() {
    super.viewDidLoad()
    
    _versionLabel?.text = "VERSION \(SpotX.version())"
    
    // create "done" button on keyboard
    keyboardDoneButtonView = UIToolbar()
    keyboardDoneButtonView.sizeToFit()
    let doneButton: UIBarButtonItem! = UIBarButtonItem(title: "Done ", style: UIBarButtonItem.Style.plain, target: self, action: #selector(doneClicked))
    let fakeButton: UIBarButtonItem! = UIBarButtonItem(barButtonSystemItem: UIBarButtonItem.SystemItem.flexibleSpace, target: nil, action: nil)
    keyboardDoneButtonView.setItems([fakeButton, doneButton], animated: false)
  }
  
  @IBAction func doneClicked(_ sender: UIButton) {
    dismissKeyboard()
  }
  
  @IBAction func backgroundTap(_ sender: UITapGestureRecognizer) {
    dismissKeyboard()
  }
  
  // Used to display a message to the user
  func showMessage(_ message: String!) {
    let alert: UIAlertController = UIAlertController(title: nil, message: message, preferredStyle: UIAlertController.Style.alert)
    
    let defaultAction: UIAlertAction = UIAlertAction(title: "OK", style: UIAlertAction.Style.default)
    alert.addAction(defaultAction)
    self.present(alert, animated: true, completion: nil)
  }
  
  // Implemented by subclasses
  func dismissKeyboard() {
  }
  
}
